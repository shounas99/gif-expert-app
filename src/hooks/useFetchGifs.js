import React, { useEffect, useState } from 'react';
import {getGifs} from '../helpers/getGifs'

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect( () => {
        //Cuerpo de mi peticion
        getGifs(category)
            .then(imgs => {
                setState({
                    data: imgs,
                    loading: false
                });
            })
    }, [category] )

    return state;
}



