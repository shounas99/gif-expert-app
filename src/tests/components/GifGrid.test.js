import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el <GifGrid />', () => {
    const category = 'Jonas Brothers';

    test('Debe mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        
    });

    test('Debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cosanum1.jpg',
            title: 'cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cosanum2.jpg',
            title: 'cosa2'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
    
    
})
