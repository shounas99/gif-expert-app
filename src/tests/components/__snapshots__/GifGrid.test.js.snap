// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    Jonas Brothers
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
